### PR TITLE
Created docs for Vue 3.x integration

### LINKED ISSUE #8135: Add docs for Vue 3 integration
## 📝 Provide a description of requested docs changes

Recently Vue 3 came out and it has some [changes in the api](https://v3.vuejs.org/guide/migration/introduction.html#breaking-changes). If you now try to follow [our integration guide](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/vuejs.html) using Vue 3, they do not match.  

So, I tried to use the new api and in `main.js` I did:
```js
import { createApp } from 'vue';
import App from './App.vue';
import CKEditor from '@ckeditor/ckeditor5-vue';

const app = createApp( App );

app.use( CKEditor );
app.mount( '#app' );
```
Then I got:
```js
[Vue warn]: `beforeDestroy` has been renamed to `beforeUnmount`. 
  at <Ckeditor editor=fn<Sb> modelValue="<p>Content of the editor.</p>" onUpdate:modelValue=fn  ... > 
  at <App>

[Vue warn]: Unhandled error during execution of render function 
  at <Ckeditor editor=fn<Sb> modelValue="<p>Content of the editor.</p>" onUpdate:modelValue=fn  ... > 
  at <App>

Uncaught TypeError: t is not a function
    at Proxy.render (ckeditor.js?3730:5)
    at renderComponentRoot (runtime-core.esm-bundler.js?5c40:673)
    at componentEffect (runtime-core.esm-bundler.js?5c40:4475)
    at reactiveEffect (reactivity.esm-bundler.js?a1e9:42)
    at effect (reactivity.esm-bundler.js?a1e9:17)
    at setupRenderEffect (runtime-core.esm-bundler.js?5c40:4458)
    at mountComponent (runtime-core.esm-bundler.js?5c40:4416)
    at processComponent (runtime-core.esm-bundler.js?5c40:4376)
    at patch (runtime-core.esm-bundler.js?5c40:3991)
    at mountChildren (runtime-core.esm-bundler.js?5c40:4180)
```
 TBH I'm not sure if there's some easy workaround for this ( ofc if I did this right ), or whether it needs some deeper changes.  
cc @oleq could you take a look at this?

### PR DESCRIPTION
**Pull request by psmyrek**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Docs: Created docs for integrations Vue.js (versions: 3.x and 2.x). Closes #8135.

### CODE REVIEW

File: docs/builds/guides/frameworks/vuejs.md [hunk 1]
```diff
@@ -80,6 +89,19 @@ const app = new Vue( {
...
+// Only for Vue 3.x
+const app = Vue.createApp( {
+	data() {
+		editor: ClassicEditor,
+		editorData: '<p>Content of the editor.</p>',
+		editorConfig: {
+			// The configuration of the editor.
+		}
+	}
+} ).use( CKEditor( Vue ) ).mount( '#app' );
```
Comment (oleq) [inline docs/builds/guides/frameworks/vuejs.md:102]: If [this line](https://github.com/ckeditor/ckeditor5/pull/8289/files#diff-b2ab47c3d98e0c613d60703d6ef661ade227658ea9062e1e0378cc853ad17904R59) explains how to enable the component globally, I think this `.use()` here is unnecessary, right?

File: docs/builds/guides/frameworks/vuejs.md [hunk 2]
```diff
@@ -154,11 +216,13 @@ If you do not want the CKEditor component to be enabled globally, you can skip t
 </template>
 
 <script>
+	const component = CKEditor( Vue ).component;
```
Comment (oleq) [inline docs/builds/guides/frameworks/vuejs.md:219]: I think this will not work in Vue 3.x.

**Review by oleq — CHANGES_REQUESTED** [review]

File: docs/builds/guides/frameworks/vuejs.md [hunk 3]
```diff
@@ -38,21 +38,29 @@ Optionally, you can [use the component locally](#using-component-locally).
...
-Vue.use( CKEditor );
+// Vue 2.x only.
+Vue.use( CKEditor( Vue ) );
+```
+
+If you have Vue 3.x, use the [application instance](https://v3.vuejs.org/guide/instance.html#creating-an-application-instance) to install the component globally:
+
+```js
+// Vue 3.x only.
+Vue.createApp( { /* options */ } ).use( CKEditor( Vue ) );
```
Comment (oleq) [inline docs/builds/guides/frameworks/vuejs.md:59]: In https://github.com/ckeditor/ckeditor5/pull/8289/files#diff-b2ab47c3d98e0c613d60703d6ef661ade227658ea9062e1e0378cc853ad17904R160 you state that `h()` is mandatory. Which way is it?

**Comment by psmyrek** [comment]
Documentation has been divided into two parts: for Vue 2.x and for Vue 3.x.

File: docs/builds/guides/frameworks/vuejs.md (hunk 1 above)
Comment (psmyrek) [reply docs/builds/guides/frameworks/vuejs.md]: The documentation has been rewritten.

File: docs/builds/guides/frameworks/vuejs.md (hunk 2 above)
Comment (psmyrek) [reply docs/builds/guides/frameworks/vuejs.md]: The documentation has been rewritten.

**Review by godai78 — APPROVED** [review]
Godai's seal of approval.

**Comment by godai78** [comment]
Wouldn't it be nice to have something similar in the Vue guides to what we have in Angular, mentioning versions?

**Supported Angular versions**
Because of the breaking changes in the Angular library output format, the ckeditor5-angular package is released in the following versions to support various Angular ecosystems:

**Versions 1.x.x – For Angular 5-8 applications**. Support for this version will end when the official support for Angular 8 is dropped (planned date: November 2020),
**Versions 2.x.x – For Angular 9.1+ applications**. This version is currently actively supported.
All available versions are listed on npm, where they can be pulled from.

**Comment by pomek** [comment]
This PR should land directly to the `#stable` branch. However, now it's too hard to rebase the stuff. I'll create the same PR starting from the `#stable` branch.

**Pull request closed without merging**